redfish-exporter/slurm/slurm.go

```diff
@@ -50,136 +53,188 @@ type SlurmServerConfig struct {
 
 type Client struct {
 	apiClient *slurmrestdapi.APIClient // slurm URL to client mapping
+	helper    map[string]reflect.Method
 }
 
-var apiCl *Client // singleton client
+var singletonAPICl *Client // singleton client
 
 func NewClient(slurmControlNode, slurmToken string) (*Client, error) {
+	c := &Client{helper: map[string]reflect.Method{}}
 	slConfig := &SlurmServerConfig{
 		URL:         slurmControlNode,
 		Username:    defaultSlurmUsername,
 		BearerToken: slurmToken,
 	}
 	cl := createRestClient(slConfig)
-	c := &Client{apiClient: cl}
+
+	// populate the methods required for ping and node update operations
+	t := reflect.TypeOf(cl.SlurmAPI)
+	postNodeRe := regexp.MustCompile(fmt.Sprintf(`%s$`, methodPostNode))
+	pingRe := regexp.MustCompile(fmt.Sprintf(`%s$`, methodPing))
+	for i := 0; i < t.NumMethod(); i++ {
+		method := t.Method(i)
+		if postNodeRe.MatchString(method.Name) {
+			postNodeExecuteMethod, found := t.MethodByName(withExecuteSuffix(method.Name))
+			if !found {
+				return nil, fmt.Errorf("could not find PostNodeExecute method from Slurm REST APIs")
+			}
+
+			if _, found := c.helper[methodPostNode]; !found {
+				c.helper[methodPostNode] = method
+				c.helper[withExecuteSuffix(methodPostNode)] = postNodeExecuteMethod
+			}
+		} else if pingRe.MatchString(method.Name) {
+			pingExecuteMethod, found := t.MethodByName(withExecuteSuffix(method.Name))
+			if !found {
+				return nil, fmt.Errorf("could not find PingExecute method from Slurm REST APIs")
+			}
+
+			if _, found := c.helper[methodPing]; !found {
+				c.helper[methodPing] = method
```

Comment:
Is it possible to store the function pointer to the method instead of storing the reflect method? That way, when calling the method you can directly call the function pointer instead of calling the function using reflect. Just a minor optimization, if possible.

Reply:
@sajmera-pensando 

```    
    // Get the function as reflect.Value
    funcValue := method.Func

    // Convert reflect.Value to an interface and type assert it as a function
    // Here, specify the function signature you expect
    funcInterface := funcValue.Interface().(func(*MyStruct, string))

    // Call the function pointer with the specific arguments
    funcInterface(myStruct, "Hello, world!")
  ````
   
   If we need to convert the reflect.Method to a function pointer then we need to know the exact API signature and In this case, the parameters are also version specific. 
   `func (a *SlurmAPIService) SlurmV0040PostNodeExecute(r ApiSlurmV0040PostNodeRequest) (*V0040OpenapiResp, *http.Response, error)`
   `func (r ApiSlurmV0040PostNodeRequest) V0040UpdateNodeMsg(v0040UpdateNodeMsg V0040UpdateNodeMsg) ApiSlurmV0040PostNodeRequest`